src/sink_http.c

```diff
@@ -87,16 +111,23 @@ static int check_elide(struct cb_info* info, char* full_name, double value) {
 }
 
 /*
- * TODO: There is a lot redundant code here with sink_stream to normalize
- * an output representation of a metrics.
+ * Callback handling add metrics. If a destination object is too full
+ * the callback will work to split the destination object and generate
+ * a new one.
  */
 static int add_metrics(void* data,
                        metric_type type,
                        char* name,
                        void* value) {
-
     struct cb_info* info = (struct cb_info*)data;
-    json_t* obj = info->jobject;
+    if (json_object_size(info->jobjects[0]) > MAX_BODY_OBJECTS) {
+        json_t** jobjects = malloc(sizeof(json_t*) * (info->jobjects_count + 1));
+        memcpy(&jobjects[1], &info->jobjects[0], sizeof(json_t*)*info->jobjects_count);
```

Comment:
I'm sure I'm missing things but a) is the jobject not still too big after shoving it down the line, and 2) what happens if jobject_count > 2

Reply:
The logic here should:

- Make a new jobjects array with size+1 slot (8 bytes)
- Copy into the new array starting at offset one from the old array (so using the original size). This leaves the 0 slot undefined.
- The line below inits the first slot in the array with a new object which will be used.

Reply:
Oh right, what even is an array